common/lib/xmodule/xmodule/tests/test_content.py

```diff
@@ -173,3 +179,101 @@ def test_static_content_write_js(self):
+            unlocked_asset_key = StaticContent.compute_location(self.new_course.id, unlocked_name)
+            unlocked_content = StaticContent(unlocked_asset_key, unlocked_name, 'image/png', unlocked_buf)
+            contentstore().save(unlocked_content)
+
+            # Create a locked image.
+            locked_buf = StringIO()
+            locked_name = "n_l.png"
+
+            # Save the course image to the content store.
+            locked_asset_key = StaticContent.compute_location(self.new_course.id, locked_name)
+            locked_content = StaticContent(locked_asset_key, locked_name, 'image/png', locked_buf, locked=True)
+            contentstore().save(locked_content)
+
+        with self.store.default_store(ModuleStoreEnum.Type.mongo):
+            self.old_course = CourseFactory.create(org='a', course='c', run='d')
+
+            # Create an unlocked image.
+            unlocked_buf = StringIO()
+            unlocked_name = "o_unl.png"
+
+            # Save the course image to the content store.
+            unlocked_asset_key = StaticContent.compute_location(self.old_course.id, unlocked_name)
+            unlocked_content = StaticContent(unlocked_asset_key, unlocked_name, 'image/png', unlocked_buf)
+            contentstore().save(unlocked_content)
+
+            # Create a locked image.
+            locked_buf = StringIO()
+            locked_name = "o_l.png"
+
+            # Save the course image to the content store.
+            locked_asset_key = StaticContent.compute_location(self.old_course.id, locked_name)
+            locked_content = StaticContent(locked_asset_key, locked_name, 'image/png', locked_buf, locked=True)
+            contentstore().save(locked_content)
+
+    @ddt.data(
+        (None, u"n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (None, u"/n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"/n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"/n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (None, u"/asset-v1:a+b+d+type@asset+block@n_unl.png", u"/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (None, u"/asset-v1:a+b+d+type@asset+block@n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/asset-v1:a+b+d+type@asset+block@n_unl.png", u"//dev/asset-v1:a+b+d+type@asset+block@n_unl.png"),
+        (u"dev", u"/asset-v1:a+b+d+type@asset+block@n_l.png", u"/asset-v1:a+b+d+type@asset+block@n_l.png"),
+        (u"dev", u"/c4x/a/b/asset/n_unl.png", u"//dev/c4x/a/b/asset/n_unl.png"),
+        (u"dev", u"/c4x/a/b/asset/n_l.png", u"//dev/c4x/a/b/asset/n_l.png"),
```

Comment:
Please test for a thumbnail block as well. You don't have to have the full set of permutations for that, just testing that thumbnail blocks actually do work would be great. A new-style thumbnail block might look something like `asset-v1:edX+DemoX+Demo_Course+type@thumbnail+block@big3-png-375x200.jpg`